D3ASIM-3218: Added command line argument for json file to be imported. (#94)
gridsingularity/d3a-api-client: README.md

## How to use the Client

### Interacting via CLI
In order to get help, please run:
```
d3a-api-client run --help
```
Following is the more in depth explanation of individual flags supported,

> - base-setup-path --> Sets the base path where you client script resides else `d3a_api_client/setups` is taken as default for your client-scripts. User could provided both an absolute and relative file path.

Reviewer: ```suggestion
- base-setup-path --> Sets the base path where your client script resides, otherwise `d3a_api_client/setups` is taken as default for your client scripts. Users can provide either an absolute or a relative file path.
```

- setup --> Name of your api-client module/script.
- username --> Username of agent authorized to communicate with respective collaboration/CN.
- password --> Password of respective agent
- domain-name --> D3A domain name
- web-socket --> D3A websocket URI

> - simulation-config-path --> Path of JSON format file that contains agent's point of interest's collaboration/CN's info (It could downloaded from the registry page of UI) 

Reviewer: I'm not sure I understood exactly what you mean here by "agent's point of interest". This file contains the information regarding the collaboration/CN, correct?

Author: `agent's point of interest` --> each users requesting it would only getting its own approved areas and selected aggregator info. May be you can propose better wording here

Reviewer: Ah, I see. Not easy...what about:
```suggestion
- `--simulation-config-path`: path to the JSON file that contains the user's collaboration/CN info (it can be downloaded from the registry page of the D3A website)
```
I think there's no need to be more specific, since you're already saying that this file will only contain information related to the user.

- run-on-redis --> This flag targets the local testing of api client, where no user authentication. A locally running redis server d3a simulation is needed here.

#### Examples

> - For local testing of api-client

Reviewer: ```suggestion
- For local testing of the API client:
```

```
d3a-api-client --log-level ERROR run --setup test_redis_aggregator --run-on-redis
```
- For testing your api client script on remote server hosting d3a's collaboration/CNs.
    - If your client script resides on `d3a_api_client/setups`
    ```
    d3a-api-client run -u <username> -p <password> --setup test_create_aggregator --simulation-config-path <your-downloaded-simulation-config-file-path>
    ```

>     - If your client script resides on any specific directory, then it needs to set via --base-setup-path

Reviewer: ```suggestion
    - If your client script resides on a different directory, then its path needs to be set via `--base-setup-path`
```
(In general, variable and parameter names should be put between single ticks in markdown)

## How to use the Client

### Interacting via CLI

> It helps to trigger your client script interact with your interested d3a's collaboration/CN conveniently via bash terminal.

Reviewer 2: I dont see the information in this sentence.

Author: Any suggestions?, else will remove it then

Reviewer 2: Yeah, remove it I would say.

> In order to get help of supported flag with its minimal explanation, please do the following

Reviewer 2: ```suggestion
In order to get help, please run:
```

```
d3a-api-client run --help
```
Following is the more in depth explanation of individual flags supported,
- base-setup-path --> Sets the base path where you client script resides else `d3a_api_client/setups` is taken as default for your client-scripts. User could provided both an absolute and relative file path.
- setup --> Name of your api-client module/script.
- username --> Username of agent authorized to communicate with respective collaboration/CN.
- password --> Password of respective agent
- domain-name --> D3A domain name
- web-socket --> D3A websocket URI
- simulation-config-path --> Path of JSON format file that contains agent's point of interest's collaboration/CN's info (It could downloaded from the registry page of UI) 

> - run-on-redis --> This flag targets the local testing of api client, where no user authentication is needed and needs to be skipped.
> 

Reviewer 2: ```suggestion
- run-on-redis --> This flag targets the local testing of api client, where no user authentication is needed. A locally running redis server d3a simulation is needed here.

```

- domain-name --> D3A domain name
- web-socket --> D3A websocket URI
- simulation-config-path --> Path of JSON format file that contains agent's point of interest's collaboration/CN's info (It could downloaded from the registry page of UI) 

> - run-on-redis --> This flag targets the local testing of api client, where no user authentication. A locally running redis server d3a simulation is needed here.

Reviewer 2: ```suggestion
- run-on-redis --> This flag targets the local testing of api client, where no user authentication. A locally running redis server and d3a simulation is needed here.
```

Reviewer: ```suggestion
- run-on-redis --> This flag targets the local testing of the API client, where no user authentication is required. A locally running redis server and d3a simulation are needed here.
```
